common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaNamingStrategyConnectNameImpl.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.amazonaws.services.schemaregistry.common;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.Schema;
+
+@Slf4j
+public class AWSSchemaNamingStrategyConnectNameImpl implements AWSSchemaNamingStrategy {
```

Comment:
Thanks for submitting this PR! 

Few comments:

1. In my opinion this is also useful for use-cases beyond Kafka connect. We can drop 'Connect' from the class name. Also, we support multiple data-formats like JSON / Protobuf (future) but logic here seems to be specific to Avro, I suggest following:
1.1 Add a TODO / issue and a note that this needs to be improved to support other data formats. I can take up the improvement later.
1.2 Feel free to implement it for JSON format if you can. :)

Name suggestion: `RecordSchemaNameNamingStrategy` or better.

2. Can we also support Avro SpecificRecord?  
3. In order to have predictable behavior, can we throw an exception if `fullName` is null or it's not an Avro record.
4. Were you able to test this end-to-end from serializer to de-serializer? It would be great if you can add a unit-test. 

Thanks again for the PR!

Reply:
Thanks for review and superb suggestions!

1. I will rename it to AWSSchemaNamingStrategyRecordSchemaNameImpl
Unfortunately it seems to me that JSON Schema converter ignores value of both **key.converter.schemaNameGenerationClass** and **value.converter.schemaNameGenerationClass** parameters. So I'm unable to overload default implementation, unfortunately.
2. Yes, this works with **key.converter.avroRecordType=SPECIFIC_RECORD** and **value.converter.avroRecordType=SPECIFIC_RECORD**
3. Will be done.
4. Currently I'm testing in my Oracle RDBMS source and sink connector environment. I will try to backport some record examples for use in end-to-end unit test.

Regards,
Aleksei

Reply:
1. Thanks! We can drop `Impl` from the name. 
2. Should the code be updated as well? From the code, it seems like it only works if the passed object is `GenericRecord`.


Thank you!

Reply:
Thanks for this PR @averemee-si !

Would it be possible for you to implement the comments and resubmit this PR ?